Automattic/pocket-casts-android
# Migrate change password screen to Jetpack Compose

## Description
- This migrates `ChangePwdFragment` to jetpack compose
- Also fixes an issue when the user updates the password and have mini player at the bottom. The UI issue was that we could not see the `Done` button because of mini player 5793da6a4868e2624edc75375617cdc481acc275
- The field form changed a bit to have the same visual as Change Email has

## Testing Instructions
1. Go to Profile -> Account -> Change password
2. Try to change your password.
3. After you changed your password ensure these events `user_password_updated ` and `account_updated_shown ` were tracked
4. Close the changed password screen.
5. Make sure `account_updated_dismissed ` was tracked
6. Back to Change password screen again
7. Try to update the password with wrong current password
8. Make sure you see the error message


## Screenshots or Screencast 

| Before | After |
|--------|--------|
| ![Screenshot_20240605_143555_Pocket Debug](https://github.com/Automattic/pocket-casts-android/assets/42220351/ac95f29f-1f2f-4de0-bf24-bba12866cda4) | ![Screenshot_20240605_143445_Pocket Debug](https://github.com/Automattic/pocket-casts-android/assets/42220351/9406ba58-1519-457b-a82b-23831be7dba6)
 | 




## Checklist
 
#### I have tested any UI changes...
